File: littleAsset/src/main/java/littleware/asset/server/db/aws/test/PackageTestSuite.java
/*
 * Copyright 2011 http://code.google.com/p/littleware
 * 
 * The contents of this file are available subject to the terms of the
 * Lesser GNU General Public License (LGPL) Version 2.1.
 * http://www.gnu.org/licenses/lgpl-2.1.html.
 */
package littleware.asset.server.db.aws.test;

import com.google.inject.Inject;
import com.google.inject.Provider;
import java.util.logging.Level;
import java.util.logging.Logger;
import junit.framework.TestSuite;
import littleware.asset.server.db.aws.AwsModule;
import littleware.asset.server.db.test.DbAssetManagerTester;
import littleware.base.Options;
import littleware.bootstrap.AppBootstrap;
import littleware.bootstrap.AppBootstrap.AppProfile;
import littleware.bootstrap.AppModule;
import littleware.bootstrap.AppModuleFactory;
import littleware.test.TestFactory;

/**
 * AWS db package tests
 */
public class PackageTestSuite extends TestSuite {
    private static final Logger log = Logger.getLogger( PackageTestSuite.class.getName() );
    
    @Inject
    public PackageTestSuite(
            AwsConnectTester connectTester,
            Provider<DbAssetManagerTester> mgrTestFactory ) {
        setName(getClass().getName());
        this.addTest(connectTester);
        this.addTest( mgrTestFactory.get() );
        this.addTest( mgrTestFactory.get().putName( "testCreateUpdateDelete" ) );
        this.addTest( mgrTestFactory.get().putName( "testHomeIdsQuery" ) );
        this.addTest( mgrTestFactory.get().putName( "testByNameQuery" ) );
    }

    /**
     * Bootstrap test suite outside of an asset server environment
     */
    public static TestSuite suite() {
        try {
            AwsModule.dbDomainOverride = Options.some( "littleTestDomain" );
            final AppBootstrap boot = AppBootstrap.appProvider.get().addModuleFactory(
                    new AppModuleFactory() {

                        @Override
                        public AppModule build(AppProfile profile) {
                            return new AwsModule(profile);
                        }
                    }).build();

            return (new TestFactory()).build(boot, PackageTestSuite.class);
        } catch ( RuntimeException ex) {
            log.log(Level.WARNING, "Failed test bootstrap", ex);
            throw ex;
        }
    }
}

File: webapp/littleApps/s3Copy/src/main/scala/littleware/apps/s3Copy/test/PackageTestSuite.scala
/*
 * Copyright 2012 http://code.google.com/p/littleware
 * 
 * The contents of this file are available subject to the terms of the
 * Lesser GNU General Public License (LGPL) Version 2.1.
 * http://www.gnu.org/licenses/lgpl-2.1.html.
 */

package littleware.apps.s3Copy
package test


import com.google.inject
import java.io
import java.util.logging.{Level,Logger}
import junit.framework.TestSuite
import littleware.asset.server.bootstrap.ServerBootstrap
import littleware.base.AssertionFailedException
import littleware.bootstrap.AppBootstrap
import littleware.test.TestFactory


class PackageTestSuite @inject.Inject()(
  pathToolTestFactory:inject.Provider[PathToolTester]
) extends TestSuite {
  setName( getClass.getName )

  Seq( 
    "testPathLs"
    ,"testPathParts"
    ,"testMimeTypes"
    ,"testPathCopy"
  ).foreach(
    (testName) => addTest( pathToolTestFactory.get.withName( testName ) )
  )
  
}

object PackageTestSuite {
  val log = Logger.getLogger( getClass.getName )

  def suite():TestSuite = try {
    log.log( Level.INFO, "Launching test suite in " + new java.io.File( "." ).getAbsolutePath + "..." )
    val suite = {
      val boot = AppBootstrap.appProvider.get().build()
      (new TestFactory()).build( boot, classOf[PackageTestSuite] )
    }
    log.log( Level.INFO, "Returning test suite to test runner ..." )
    suite
  } catch {
    case ex:Throwable => {
        log.log( Level.WARNING, "Failed to launch test suite", ex )
        throw new AssertionFailedException( "Failed to launch test", ex )
      }
  }
  
}